File: backend/app/schemas/nearby_hospitals.py
from pydantic import BaseModel, Field
from typing import List, Optional

class UserLocation(BaseModel):
    latitude: float
    longitude: float

class NearestHospitalsRequest(BaseModel):
    latitude: float
    longitude: float
    radius_km: float = Field(default=5.0, ge=0.5, le=50.0)
    max_results: int = Field(default=10, ge=1, le=50)
    hospital_type: Optional[str] = None
    language: str = "en-IN"
    speaker: str = "anushka"

class NearbyHospital(BaseModel):
    id: int
    name: str
    hospital_type: str
    latitude: float
    longitude: float
    address: Optional[str] = None
    contact: Optional[str] = None
    rating: Optional[float] = None
    distance_km: float
    distance_display: str
    estimated_time_min: int
    estimated_time_display: str

class RouteStep(BaseModel):
    instruction: str
    distance_m: float
    duration_s: float
    name: str

class RouteDetails(BaseModel):
    total_distance_km: float
    total_distance_display: str
    total_duration_min: float
    total_duration_display: str
    route_geometry: List[List[float]]
    steps: List[RouteStep]
    google_maps_url: str
    apple_maps_url: str

class NearestHospitalsResponse(BaseModel):
    success: bool
    user_location: UserLocation
    hospitals: List[NearbyHospital]
    total_found: int
    radius_km: float
    voice_response: Optional[dict] = None
    text_response: str

class DirectionsResponse(BaseModel):
    success: bool
    hospital: NearbyHospital
    route: RouteDetails
    voice_response: Optional[dict] = None
    text_response: str

File: frontend/src/types/nearbyHospitals.ts
export interface UserLocation {
  latitude: number;
  longitude: number;
  accuracy?: number;
}

export interface NearbyHospital {
  id: number;
  name: string;
  hospital_type: string;
  latitude: number;
  longitude: number;
  address?: string;
  contact?: string;
  rating?: number;
  distance_km: number;
  distance_display: string;
  estimated_time_min: number;
  estimated_time_display: string;
}

export interface RouteStep {
  instruction: string;
  distance_m: number;
  duration_s: number;
  name: string;
}

export interface RouteDetails {
  total_distance_km: number;
  total_distance_display: string;
  total_duration_min: number;
  total_duration_display: string;
  route_geometry: [number, number][];
  steps: RouteStep[];
  google_maps_url: string;
  apple_maps_url: string;
}

export interface NearestHospitalsResponse {
  success: boolean;
  user_location: UserLocation;
  hospitals: NearbyHospital[];
  total_found: number;
  radius_km: number;
  voice_response?: { audio_base64: string; format: string; language: string };
  text_response: string;
}

export interface DirectionsResponse {
  success: boolean;
  hospital: NearbyHospital;
  route: RouteDetails;
  voice_response?: { audio_base64: string; format: string; language: string };
  text_response: string;
}
